lxd/storage/load.go

```diff
@@ -0,0 +1,120 @@
+package storage
+
+import (
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/lxd/storage/drivers"
+	"github.com/lxc/lxd/shared/api"
+)
+
+// MockBackend controls whether to run the storage logic in mock mode.
+var MockBackend = false
+
+// CreatePool creates a new storage pool on disk and returns a Pool interface.
+func CreatePool(state *state.State, poolID int64, dbPool *api.StoragePool, op *operations.Operation) (Pool, error) {
+	// Sanity checks.
+	if dbPool == nil {
+		return nil, ErrNilValue
+	}
+
+	// Ensure a config map exists.
+	if dbPool.Config == nil {
+		dbPool.Config = map[string]string{}
+	}
+
+	// Handle mock requests.
+	if MockBackend {
+		pool := mockBackend{}
+		pool.name = dbPool.Name
+		pool.state = state
+		return &pool, nil
+	}
+
+	// Define a function to retrieve a volume ID for a volume Name for use in driver.
+	volIDFunc := func(project string, volType drivers.VolumeType, volName string) (int64, error) {
```

Comment:
The drivers shouldn't have the concept of projects at all, so if that function is to be exposed there, that's going to be a problem.

Reply:
@stgraber the concept of projects appears to be hardcoded as "default" in the current storage layer. I can certainly hardcode "default" into the volIDFunc itself so that its not needed as a param, however it would be good to understand how projects will work with the storage layer in the future, as the volIDFunc is defined when the storage pool is instantiated and so would work fine if the storage pool was linked to a project, but wouldn't work if the project was used on a per-volume basis.

Reply:
@stgraber as discussed, the volName will contain the project name in the future separated by an underscore.